app/screens/home/recent_mentions/index.tsx

```diff
@@ -1,24 +1,110 @@
-    const params = route.params! as {direction: string};
-    const toLeft = params.direction === 'left';
-    const [start, setStart] = useState(false);
+import {getRecentMentions} from '@actions/remote/search';
+import DateSeparator from '@app/components/post_list/date_separator';
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import {SystemModel, UserModel} from '@app/database/models/server';
+import NavigationHeader from '@components/navigation_header';
+import {LARGE_HEADER_PROGRESSVIEW_OFFSET} from '@constants/view';
+import {useServerUrl} from '@context/server_url';
+import {withTheme} from '@context/theme';
+import {useCollapsibleHeader} from '@hooks/header';
 
-    const onLayout = useCallback(() => {
-        setStart(true);
-    }, []);
+import Mention from './components/mention';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type PostModel from '@typings/database/models/servers/post';
+
+const AnimatedSectionList = Animated.createAnimatedComponent(SectionList);
+
+const {USER, SYSTEM, POST} = MM_TABLES.SERVER;
+
+type Props = {
+    mentions: PostModel[];
+    theme: Theme;
+    currentUser: UserModel;
+}
+
+type Section = {
+    key: string;
+    title: string;
+    data: PostModel[];
+    keyExtractor: (item: PostModel) => string;
+}
+
+const styles = StyleSheet.create({
+    loading: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+});
+
+function getSections(items: PostModel[]) {
+    const groups = groupBy(items, (mention) => {
+        return moment(mention.createAt).startOf('day').format();
+    });
+
+    const sections: Section[] = Object.keys(groups).map((key) => ({
+        key: String(key),
+        title: String(key),
+        data: groups[key],
+        keyExtractor: (item: PostModel) => item.id,
+    }));
+
+    return sections.sort((a, b) => (a.title > b.title ? -1 : 1));
+}
+
+const RecentMentionsScreen = ({mentions, theme, currentUser}: Props) => {
+    const {formatMessage} = useIntl();
+    const [refreshing, setRefreshing] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const searchScreenIndex = 1;
+
+    const subtitle = formatMessage({id: 'screen.mentions.subtitle', defaultMessage: 'Messages you\'ve been mentioned in'});
+    const title = formatMessage({id: 'screen.mentions.title', defaultMessage: 'Recent Mentions'});
+    const isLargeTitle = true;
+    const isFocused = useIsFocused();
+    const {scrollPaddingTop, scrollRef, scrollValue, onScroll} = useCollapsibleHeader<SectionList<string>>(isLargeTitle, Boolean(subtitle), false);
+    const paddingTop = useMemo(() => ({paddingTop: scrollPaddingTop, paddingBottom: 10}), [scrollPaddingTop]);
+
+    const serverUrl = useServerUrl();
+    const sections = useMemo(() => getSections(mentions), [mentions]);
+
+    useEffect(() => {
+        setLoading(true);
+        getRecentMentions(serverUrl).finally(() => {
```

Comment:
So that we can have mentions appear even when offline. 
By edit you mean removing the mention keyword? I don't necessarily think that it would cause a problem, but you are right we must handle a deleted post, at least to not crash the app.

Reply:
Maybe I am missing some of the logic. Feel free to correct me if I am wrong.

getRecentMentions fetch from the server, and store things on the database. You are not doing a local mention search in the local database. Or are you doing both?

Reply:
@larkox So the idea is that if you fetched some mentions in the past and then you go offline and switch to the recent mentions screen, it will know what posts (that you already have in the db) you where mentioned in and then load those. (they will also show up while your performing a new request to the server)

Reply:
Oh, that sounds cool. I get it now. LGTM then.